src/librustc_driver/driver.rs

```diff
@@ -378,9 +380,11 @@ impl<'a, 'b, 'ast, 'tcx> CompileState<'a, 'b, 'ast, 'tcx> {
                          krate: ast::Crate,
                          cstore: &'a CStore)
                          -> CompileState<'a, 'b, 'ast, 'tcx> {
+        let cloned_span = krate.span.clone();
         CompileState {
             krate: Some(krate),
             cstore: Some(cstore),
+            registry: Some(Registry::new(&session, cloned_span)),
```

Comment:
I believe this can be just `registry: Some(Registry::new(&session, krate.span)),`.